src/pipelinedefinition.cpp

```diff
@@ -384,4 +384,63 @@ Status PipelineDefinition::getInputsInfo(tensor_map_t& inputsInfo, ModelManager&
     return StatusCode::OK;
 }
 
+Status PipelineDefinition::getOutputsInfo(tensor_map_t& outputsInfo, const ModelManager& manager) const {
+    // Assumptions: this can only be called on available pipeline definition.
+    // Add check if available when pipeline status will be implemented.
+
+    static const auto byName = [](const std::string& name) {
+        return [name](const NodeInfo& nodeInfo) {
+            return nodeInfo.nodeName == name;
+        };
+    };
+
+    for (const auto& [dependant_node_name, all_mappings] : connections) {
+        const auto& dependant_node_info = std::find_if(std::begin(nodeInfos), std::end(nodeInfos), byName(dependant_node_name));
+        if (dependant_node_info->kind != NodeKind::EXIT) {
+            continue;
+        }
+
+        for (const auto& [dependency_node_name, specific_dependency_mapping] : all_mappings) {
+            const auto& dependency_node_info = std::find_if(std::begin(nodeInfos), std::end(nodeInfos), byName(dependency_node_name));
+
+            switch (dependency_node_info->kind) {
+            case NodeKind::ENTRY: {
+                for (const auto& [alias, real_name] : specific_dependency_mapping) {
+                    outputsInfo.insert({real_name, TensorInfo::getUnspecifiedTensorInfo()});
+                }
+                break;
+            }
+            case NodeKind::DL: {
+                auto instance = manager.findModelInstance(dependency_node_info->modelName, dependency_node_info->modelVersion.value_or(0));
+                if (!instance) {
+                    // TODO: Change to SPDLOG_DEBUG before release
+                    SPDLOG_INFO("Model:{} was unavailable during pipeline:{} outputs info fetching", dependency_node_info->modelName, this->getName());
+                    return StatusCode::MODEL_MISSING;
+                }
+                std::unique_ptr<ModelInstanceUnloadGuard> unloadGuard;
+                auto status = instance->waitForLoaded(0, unloadGuard);
```

Comment:
Maybe add test with mocked model instance which will perform long load (1s), and we will launch during that getInputs/outputs info and check that we do not wait for model to load? Eg check execution time with small threshold.

Reply:
I have added test for calling GetInputsInfo/GetOutputsInfo when one of models is in LOADING state. Does it cover your suggestion?
https://github.com/openvinotoolkit/model_server/pull/337/files#diff-171cd48347cc255c38e835edff0458e1008d80517d18f922106abf9445074c6cR329